### PR TITLE
Split `dashboard.tsx` into smaller components

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
See commit messages for a more comprehensive list of changes.
- Split `dashboard.tsx` into more components: projects table, directories table, secrets table, files table, column
  - Closes https://github.com/enso-org/cloud-v2/issues/401

#### New features
- Multi-select (ctrl+click and shift+click should work like on Windows/Linux)
- Delete multiple assets using context menu (right click) (when using multi-select)
  - Closes [cloud-v2/#479](https://github.com/enso-org/cloud-v2/issues/479)
- Use project names instead of IDs in error messages. Using IDs was intentional as cloud assets may have the same name, however in the vast majority of cases using the name will be sufficient, look better, and be more understandable.
- Inline editing + tick and cross icons when renaming items in a folder
  - Addresses part of [cloud-v2/#489](https://github.com/enso-org/cloud-v2/issues/489):
  - Ctrl+click and select + single click (NOT double click) should both work, to begin editing.
  - Probably out of scope, but it depended on this refactor anyway
- Proper optimistic updates - pending inserts show as `opacity-50` (for now at least), pending deletes are hidden but still in the items list

#### Debugging improvements
- Add React hooks for debugging
- Adds support for `assert` - this is currently unused as I'm not sure where an assert would be useful.
- Minimize re-renders - this is *not* a performance optimization, but it helps with debugging

### Screencasts
Delete multiple, rename, optimistic deletes/inserts:

[screen-recorder-tue-jul-11-2023-17-56-07.webm](https://github.com/enso-org/enso/assets/4046547/0226e7df-719b-461f-ac44-0dd7bb15d4ac)

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by somebody1234** [comment]
Note that many issues with missing docs and sections should be fixed in #6403

**Comment by somebody1234** [comment]
Manually requesting review while this is still in a draft state, since this is only blocking on bugfix PRs. It should be doable to drop the bugfix PR commits from the history, but I'd prefer not to risk more nasty merge conflicts

**Comment by somebody1234** [comment]
dang, the commit times are all messed up :(

**Review by wdanilo — APPROVED** [review]

**Comment by sylwiabr** [comment]
@somebody1234 what is the status here?

**Comment by somebody1234** [comment]
@sylwiabr this one will be merged in after all other dashboard PRs because it is a big change. it's also waiting on QA, but i think that's fine because it will likely need to be merged a few times as the other PRs get merged in

**Comment by wdanilo** [comment]
@somebody1234 if a PR is waiting for other PRs, please keep it as draft :)

**Comment by PabloBuchu** [comment]
@somebody1234 can we fix the confflicts and merge latest develop here?

**Comment by somebody1234** [comment]
there are issues unfortunately (at least one) :(
the current issue is that the IDE is completely broken. this is most likely because of react's strict mode, which mounts all components (including the IDE component) twice... i'll investigate later, but i think it's better to implement more qol improvements before coming back to this PR.

**Comment by somebody1234** [comment]
cherry-picked some commits on a branch that i had lying around locally, so that i can delete the local branch. i figured this is the best place to put them

**Comment by somebody1234** [comment]
currently testing and fixing bugs. note that some things *will* display twice because of strict mode in, when in dev mode (anything but `ide build`). *this is intentional*. (should definitely check that they only display once outside of strict mode though)

currently the only place i've seen this happen is with an invalid `-startup.project` though.
it *is* also causing issues with the IDE opening though - since React is trying to open the IDE twice
(edit: IDE issues "fixed" by detecting strict mode and handling it appropriately. a comment has been added to `ide.tsx` explaining why I don't think there's a better solution (unfortunately))

**Comment by somebody1234** [comment]
note: some local commits incoming soon

**Comment by somebody1234** [comment]
there are some fixes in there, a new feature or two, and a lot of re-render optimization - not necessarily for performance (since the official React stance is that it doesn't really improve performance) - but it should make debugging better, as each change will not result in the *entire* dashboard re-rendering

**Comment by somebody1234** [comment]
rename failure tested by changing `!responseIsSuccessful(response)` to `!responseIsSuccessful(response) || Math.random() > 0.5` in `projectUpdate` in `remoteBackend.tsx`

**Comment by somebody1234** [comment]
i think this is split enough - i am not against splitting further, but i'm not sure whether there's a place where it makes sense to split further.

instead, for convenience, here are groups of commits. checkboxes are included for ease of copying, in case it is easier to check each individual group off as they are reached:
- [ ] the initial group of commits. this contains a lot of content that has been split into other PRs <https://github.com/enso-org/enso/pull/6546/files/c2642404092eedec807c9ce748d7494d33109b44#diff-3a6cb83548cd7c775db1f9fd059f63b99e6213e91ae65d4017185bd2abf1710e>
  - [ ] here is a temporary branch made, excluding the content that has been split into other PRs: <https://github.com/enso-org/enso/compare/b0ce2a8d298a2639842ba771d2ba64947556af05..b6ca2ff3c4111897a67266704e3cbe856f19e3da>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/2047770fec8de1649e91b51d6e1472244526fd46>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/19f407b6e12a65f9e9f9fb721a4c8eee7da68a41> (very tiny)
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/66d1be30415dce59178cdb85d1e4c2d45e68c4bf> (very tiny)
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/f8a069d5722ba6afbb1933908b45077c78af2522> (very tiny)
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/6b55830f7e04715c4fdce53b007e6428ce47543c>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/7d01213cb7d3d227bc13e587af981a1042b632aa> (very tiny)
- [ ] <https://github.com/enso-org/enso/pull/6546/files/cacc8a77a3512073b5e4f9bace38244e5bdb7f7b..223929395474e238e749abfbd631778f94bce4b1>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/f652b2b7e9b437c5467aad84e4ad69297f32abc4>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/6649d8b0e07d2407dd4827c71b5d546bd024c22d> (very tiny)
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/74876bab421967c80c22a2f699bff72a63f09059> (very tiny)
- [ ] <https://github.com/enso-org/enso/pull/6546/files/dc0b30792cd58984ba8e12a74a8d8ae6861b9628..103564d569d78354647e0137e6d16f94a5821445>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/b6d7d65ab66714af66d5d552acf4530ea5ffab6c>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/3471c60a661c84c96b75d1123e334dc08d7fce8f>
- [ ] <https://github.com/enso-org/enso/pull/6546/commits/10f727278a67b80ce91d3d59f93a12083030da4d>
- [ ] <https://github.com/enso-org/enso/pull/6546/files/10f727278a67b80ce91d3d59f93a12083030da4d..9ee63483b6691ff6a89d32e1a54bcff46f716886>

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/pull/6546">Split `dashboard.tsx` into smaller components by somebody1234 · Pull Request #6546 · enso-org/enso</a></strong></div><div>⚠️ This PR depends on #7023, and MUST NOT be merged in before that one.
Pull Request Description
See commit messages for a more comprehensive list of changes.

Split dashboard.tsx into other files
...</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Comparing b0ce2a8d298a2639842ba771d2ba64947556af05..b6ca2ff3c4111897a67266704e3cbe856f19e3da · enso-org/enso</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/pull/6546">Split `dashboard.tsx` into smaller components by somebody1234 · Pull Request #6546 · enso-org/enso</a></strong></div><div>⚠️ This PR depends on #7023, and MUST NOT be merged in before that one.
Pull Request Description
See commit messages for a more comprehensive list of changes.

Split dashboard.tsx into other files
...</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/pull/6546">Split `dashboard.tsx` into smaller components by somebody1234 · Pull Request #6546 · enso-org/enso</a></strong></div><div>⚠️ This PR depends on #7023, and MUST NOT be merged in before that one.
Pull Request Description
See commit messages for a more comprehensive list of changes.

Split dashboard.tsx into other files
...</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso/pull/6546">Split `dashboard.tsx` into smaller components by somebody1234 · Pull Request #6546 · enso-org/enso</a></strong></div><div>⚠️ This PR depends on #7023, and MUST NOT be merged in before that one.
Pull Request Description
See commit messages for a more comprehensive list of changes.

Split dashboard.tsx into other files
...</div></blockquote>

**Comment by wdanilo** [comment]
@somebody1234 oh wow, you split it to really a lot of things there. I was thinking about just 2-3 PRs when talking with greg, as it takes us really long to deliver. Anyway, The best way to review things is by having just open, ready for review PRs - could you just open PRs ready for review, please?

**Comment by somebody1234** [comment]
@wdanilo it is 2-3 PRs, but this main PR is still quite large so I think this one might still be quite difficult to review. the issue with these PRs is they need to be merged sequentially.

it's probably possible to split them apart but i don't think it's worth the potential merge conflicts.

that said i don't think the first two will be changed further so i guess it's safe enough to open them?

**Comment by wdanilo** [comment]
@somebody1234 I don't know what PRs are you referring to – you've written that it is 2-3 PRs, where are they? I can't find them. We really need to merge it. We are always trying to work on a PR max 1 week, preferably shorter. This one is being developed for 6-7 weeks already. Please provide me with links to these PRs so we can review them and merge them as soon as possible :)

**Review by wdanilo — CHANGES_REQUESTED** [review]
This code changed a lot since my last review, so marking it as "to be reviewed".

File: app/ide-desktop/lib/content/src/index.ts
Comment (wdanilo) [inline app/ide-desktop/lib/content/src/index.ts]: All logs should be sentences. Please fix it in all similar places.

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx]: What do you mean by "must not be a dependency" - dependency to what? Also, I think we can write it a little bit more professionally, something along the lines of "We are not using `userSession` here because ...". I know this is a very minor thing, but we are trying to keep the comments in the code not using such descriptions as "ugly back" :P

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx]: I don't think this comment should be in code.

File: app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx]: There should be more information of why something like that happens - why it will loop forever?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/backend.ts]: Helpers / Utils - are names that do not really tell what's inside and very fast become containers for all kind of garbage. We try not to use these names anywhere. Could you please categorize these "utils" with more precise names, please?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx]: shortcuts should be defined / configurable from within one file. Otherwise it would be very hard to change them and allow for overriding them by users in the future.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx]: This is magical Can we refactor such places in code to well named variables, like `const whatItReallyChecks = event.detail === 1`, please?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx]: Does it mean that this is buggy implementation? Or this if FIXME? If the former, we need to fix it before merge. If the later, it should have FIXME comment style.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoryView.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoryView.tsx]: I don't understand this comment. Would you mind expanding it, please?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoryView.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoryView.tsx]: If this is unsafe, why are we using it? In case any unsafe function is used, it is very important to always provide safety guarantees, so the code reader will know that the code will not break.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx]: a lot of repeated code. Can we refactor the "s"-adding function somewhere, please? :)

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx]: Minor - this comment is split into multiple lines although not all of the space is used.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx]: Do we need to have circular imports? We never did circular imports in our Rust GUI codebase, as they are really hard to read. Do we need to do them here?

If we need, why are we placing such a warning? I'm just asking what is its purpose.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx]: Comments should be sentences - dot missing.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/secretsTable.tsx
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/secretsTable.tsx]: TODOs should have tasks assigned and information what wrong happens now. Is there any bad effect that we are not waiting for backend? It should be described here.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/localBackend.ts
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/localBackend.ts]: The same comment as above - if its unsafe, we should either remove it, or write safety guarantees.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/validation.ts
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/validation.ts]: What are "local projects"? If they are "not cloud projects", why we have different validation for local and cloud project names?

File: app/ide-desktop/lib/content/src/index.ts
Comment (somebody1234) [reply app/ide-desktop/lib/content/src/index.ts]: this was left in for debugging, it has been removed locally

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx]: this is also fixed locally, will push soon

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/validation.ts
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/validation.ts]: cloud projects (currently) have no restrictions on project name. local projects have a restriction put in place by the Project Manager.

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/directoriesTable.tsx]: unfortunately i think this is just a side effect of how react works. realistically it won't be a problem for directories under, say, a couple thousand entries, but the intention is to have the comment here so anyone looking at the code is aware that this is a good place to start looking into performance optimizations when performance does become a problem

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx]: on a completely unrelated note it was originally supposed to say "ugly hack" but i can't type

File: app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/components/app.tsx]: upon inspection it turns out that `userSession` can be removed from the dependencies in `SessionProvider` and the linter doesn't complain. not sure why it was added to the dependencies in the first place tbh... maybe i added it there by accident

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx]: done locally. it's worth noting that now the definition of every `TOAST_PROMISE_MULTIPLE_MESSAGES` is virtually identical, but i'd be cautious about refactoring that too, as it will likely result in a file being created for just the one function - unless you have any ideas?

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx]: i think the first line has to stay as it is - only `However,` would fit from the second line

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/filesTable.tsx]: side note: it's worth noting that the word "UNSAFE" is used here - however it's technically not *unsafe*, but rather it's just usually bad practice. it's probably a good idea to change the wording, but i'm not quite sure what may be best. maybe something like "is a bad practice" or "is usually a bad practice"

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx]: i'll move `ProjectActionButton` into `projectsTable.tsx`

as for the warning: it's not strictly needed, but i have a feeling that it would potentially cause problems if we ever stop bundling

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/validation.ts
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/validation.ts]: comment added. (the reason why cloud projects don't have the same restriction is because the name in the database is not related to the name passed to the cloud VM's Project Manager - which i believe is always `Project_root`)

File: app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/projectActionButton.tsx]: on second thought, it's too long to inline, i'll just extract `ProjectRowState` to its own file instead.

side note: the reason i went with a circular import is because it's pretty awkward to extract `ProjectRowState` to its own file. but of course, it's better than nothing

**Comment by somebody1234** [comment]
there are still some minor issues left to fix

**Comment by somebody1234** [comment]
oh... deleting projects being disabled isn't checked for multiple delete... but row-specific state is currently local to each row

**Comment by indiv0** [comment]
Seeing lots of conflicts on this PR, delaying QA+review until they're addressed.

**Comment by xvcgreg** [comment]
@somebody1234 what's the holdup here?

**Comment by somebody1234** [comment]
@xvcgreg Still blocked by #7023 - I guess I'll mark as draft again for the time being

**Comment by wdanilo** [comment]
Ok, guys, we need to merge all of these issues sooner than later. @PabloBuchu what is the plan here? Can we merge all of them this week?

**Comment by somebody1234** [comment]
even more changes
:( unfortunately i think many of these are required for optimistic UI to work at all
most things should work fine, but there are still a couple minor issues:
- [ ] entries flash solid then back to `opacity-50` when you're adding entries too quickly (e.g. spam adding directories)
- [x] `projectActionButton` should be merged back into `projectsTable` to ~~greatly~~ simplify logic
  - ~~this should make `rowState` obsolete~~ `projectActionButton` still needs to access the state somehow
- [x] `RenameModal` (currently opened by the context menu) should also be removed in favor of editable row names.
- [ ] this eventually needs to be updated to optimistically update state when the "shared with" modal is submitted. (this depends on the "shared with" modal being merged in, of course, so it is currently not possible.)

**Comment by wdanilo** [comment]
@PabloBuchu ping

**Comment by PabloBuchu** [comment]
@wdanilo yes plan for today is to test and hopefully merge:
https://github.com/enso-org/enso/pull/7023 
https://github.com/enso-org/enso/pull/7128
https://github.com/enso-org/enso/pull/7145

so it will clean a bit the PR board and renable this PR for qa and review

**Comment by somebody1234** [comment]
heads up: there's a huge amount of duplication between the assets tables (`projectsTable.tsx`, `directoriesTable.tsx` etc). it's definitely possible to refactor that out, but i'd first like to make sure that's something we want to do

**Comment by PabloBuchu** [comment]
> heads up: there's a huge amount of duplication between the assets tables (`projectsTable.tsx`, `directoriesTable.tsx` etc). it's definitely possible to refactor that out, but i'd first like to make sure that's something we want to do

so after we merged the 7023 lets take care and open this PR for review. Later we will start thinking on refactoring `projectsTable` and `directoriesTable`

**Comment by somebody1234** [comment]
seems like there have been no logic changes to `dashboard.tsx` since [the last merge with `develop`](https://github.com/enso-org/enso/pull/6546/commits/d1332946ef5d3fcf83d783f4774c7fdc943cf456): <https://github.com/enso-org/enso/compare/8504295e5e9f106e3aa039a93a50563a9fa5d0be..7749286c6964df3d0417f6c5dce991535f2a98f5>

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Comparing 8504295e5e9f106e3aa039a93a50563a9fa5d0be..7749286c6964df3d0417f6c5dce991535f2a98f5 · enso-org/enso</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by PabloBuchu** [comment]
@somebody1234 are we ready to merge in latest develop, fix the description and open for review?

**Comment by wdanilo** [comment]
Guys, something is really broken with this PR. It was opened on May 5th ... Please talk with each other and make sure something like that does not happen in the future. Anyway, we need to close it ASAP.

**Comment by PabloBuchu** [comment]
qa: 
- multiple select works partially on mac. shift+click works (selects range) but ctrl/cmd + click does not
- when I leave the rename input it will go back to being a label in random time
- when in rename project mode double click on project should not open it. it makes impossible for user to double click on the project name to select and copy 
- got a warning when I was registering and setting new username
Screenshot 2023-07-11 at 14 11 16

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx
Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx]: whats the difference between while and empty for loop?

Comment (PabloBuchu) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx:256]: what is the difference between empty for and while?

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx]: none, it's just that lints disallow `while (true)` (?!?!)
... actually yeah this is confusing, i'll just configure the rule to prefer `while (true)` over this

**Review by indiv0 — COMMENTED** [review]
Code LGTM. Didn't perform QA. Looks like there's some failing checks & merge conflicts?

**Review by indiv0 — APPROVED** [review]

**Review by PabloBuchu — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]

**Pull request merged**
